M2/cmake/configure.cmake

```diff
@@ -62,26 +62,19 @@ set(PACKAGE_VERSION ${Macaulay2_VERSION})
 find_package(Git QUIET)
 if(GIT_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/../.git")
   execute_process(
-    COMMAND ${GIT_EXECUTABLE} describe --dirty --always --match HEAD
+    COMMAND ${GIT_EXECUTABLE} describe --tags --dirty
     ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-    OUTPUT_VARIABLE   GIT_COMMIT)
-  # TODO: currently finds the last commit that changed the VERSION file
-  # but ideally it should get the last release commit instead
+    OUTPUT_VARIABLE   GIT_DESCRIPTION)
   execute_process(
-    COMMAND ${GIT_EXECUTABLE} rev-list -1 HEAD VERSION
+    COMMAND ${GIT_EXECUTABLE} branch --show-current
```

Comment:
Turns out if you're on a detached state (say if you checkout `upstream/development` or someone else's branch) this is just an empty string, which shows up as `()` in the version string. An alternative might be `git show -s --pretty=%D HEAD` but it's a bit verbose. Here are a couple of examples:
```
HEAD -> SheafMaps, origin/SheafMaps
HEAD, tag: release-1.24.05, upstream/release-1.24.05-branch, upstream/master
```

Reply:
If we're not on a branch, then getting the empty string for `version#"git branch"` seems like reasonable behavior.  We have `version#"git description"` to tell us more already.

Maybe just don't include the branch info in the startup banner if it's empty?

Reply:
That seems reasonable.

Also, I realized a major drawback of including anything more than just the version in the binary is that any change in git will cause M2-binary to recompile. It's not urgent to change this, but I think a possible solution is to move the most frequently changed numbers into a non-compiled file (e.g. a variable set in an m2 file).